default/scripting/buildings/buildings.py

```diff
@@ -132,3 +138,29 @@
         ],
     ),
 ]
+
+
+CAN_ADD_STARLANE_TO_SOURCE = (
+    System
+    & ~Contains(IsSource)
+    & ~HasStarlane(from_=Object(id=Source.SystemID))
+    # specify the system, not the source object. the condition will consider multiple lanes
+    # connecting to a system as not crossing,
+    # but if the object is not itself a system but is in a system,
+    # then the system may be considered as crossing a lane from the object's location
+    & ~StarlaneToWouldCrossExistingStarlane(condition=Object(id=Source.SystemID))
+    & ~StarlaneToWouldBeAngularlyCloseToExistingStarlane(condition=Object(id=Source.SystemID), maxdotprod=0.87)
+    & ~StarlaneToWouldBeCloseToObject(
+        distance=20.0,
+        condition=Object(id=Source.SystemID),
+        closeto=(
+            System  # don't go near other systems
+            & ~IsSource  # source is probably a building or planet, exclude any planet or system at the location of the source, but don't assume that the source is in a system or is on a planet when doing so
+            & ~Contains(scope=IsSource)
```

Comment:
It accepts single argument by position not by name like in `ContainedBy`

Reply:
Why is the parameter named `scope`?